src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx

```diff
@@ -0,0 +1,75 @@
+import { formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { SpellLink } from 'interface';
+import UptimeIcon from 'interface/icons/Uptime';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import { STATISTIC_ORDER } from 'parser/ui/StatisticBox';
+import TALENTS from 'common/TALENTS/warlock';
+
+class FelCovenant extends Analyzer {
+  static talent = TALENTS.FEL_COVENANT_TALENT;
+  static buffId = SPELLS.FEL_COVENANT_BUFF.id;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(FelCovenant.talent);
+
+    if (!this.active) {
```

Comment:
This `if` check isn't necessary

Reply:
Removed it. It does appear in a lot of other analyzers though.

Reply:
We've had mixed messaging on it in the past, so a lot of them have it hanging around